test/integration/rmi_out_test.py

```diff
@@ -40,32 +40,45 @@ def test_pudl_engine(pudl_engine, table_name):
 def test_ppl_out(rmi_out, request):
     """Test generation of the EIA Plant Parts List."""
     clobber = not request.config.getoption("--cached-plant-parts-eia")
-    ppl = rmi_out.plant_parts_eia(clobber=clobber)
+    ppl = rmi_out.plant_parts_eia(
+        clobber=clobber, pickle_distinct=True, pickle_train_connections=True
+    )
     assert not ppl.empty
-    for ppl_df in ["plant_parts_eia", "gens_mega_eia", "true_grans_eia"]:
-        if ppl_df in rmi_out.pudl_out._dfs:
-            del rmi_out.pudl_out._dfs[ppl_df]
+    del ppl
+    df_keys = list(rmi_out.pudl_out._dfs.keys())
```

Comment:
After each of these tests I clear `pudl_out._dfs`. Maybe there's a better way to do this.

Reply:
this will definitely mean that the tests re-process several `pudl_out` tables. We definitely use like the generator and utility tables in various steps along the way. But I suppose this is worth it... If it is possible memory-wise, I could imagine doing this for just the ppe outputs because they are already pickled and accessed this way in this repo.

But if this remove all `pudl_out` cached dfs step is needed, I'd add a lil module helper function that does this so you don't have to copy/paste it in all of these tests.

Reply:
We're less memory constrained now so I took this clearing of `pudl_out._dfs` out.